core/components/login/model/login/login.class.php

```diff
@@ -185,18 +185,8 @@ public function sendEmail($email,$name,$subject,$properties = array()) {
      * @param integer $length The length of the generated password.
      * @return string The newly-generated password.
      */
-    public function generatePassword($length=8) {
-        $pword = '';
-        $charmap = '0123456789bcdfghjkmnpqrstvwxyz';
-        $i = 0;
-        while ($i < $length) {
-            $char = substr($charmap, rand(0, strlen($charmap)-1), 1);
-            if (!strstr($pword, $char)) {
-                $pword .= $char;
-                $i++;
-            }
-        }
-        return $pword;
+    public function generatePassword($length=null) {
+        $this->modx->user->generatePassword($length);
```

Comment:
It seems that this change causes issues on one installation. The lp property is empty there during register.